backends/occa/ceed-occa-elem-restriction.cpp

```diff
@@ -0,0 +1,435 @@
+      delete [] transposeDofOffsets_h;
+      delete [] transposeDofIndices_h;
+    }
+
+    void ElemRestriction::setupKernelBuilders() {
+      ::occa::properties kernelProps;
+      kernelProps["defines/CeedInt"]    = ::occa::dtype::get<CeedInt>().name();
+      kernelProps["defines/CeedScalar"] = ::occa::dtype::get<CeedScalar>().name();
+
+      kernelProps["defines/COMPONENT_COUNT"] = ceedComponentCount;
+      kernelProps["defines/ELEMENT_SIZE"]    = ceedElementSize;
+      kernelProps["defines/TILE_SIZE"]       = 64;
+      kernelProps["defines/USES_INDICES"]    = usesIndices();
+
+      applyKernelBuilder = ::occa::kernelBuilder::fromString(
+        occa_elem_restriction_source, "applyRestriction", kernelProps
+      );
+
+      applyTransposeKernelBuilder = ::occa::kernelBuilder::fromString(
+        occa_elem_restriction_source, "applyRestrictionTranspose", kernelProps
+      );
+    }
+
+    ElemRestriction* ElemRestriction::from(CeedElemRestriction r) {
+      if (!r || r == CEED_ELEMRESTRICTION_NONE) {
+        return NULL;
+      }
+
+      int ierr;
+      ElemRestriction *elemRestriction;
+
+      ierr = CeedElemRestrictionGetData(r, (void**) &elemRestriction);
+      CeedOccaFromChk(ierr);
+
+      ierr = CeedElemRestrictionGetCeed(r, &elemRestriction->ceed);
+      CeedOccaFromChk(ierr);
+
+      return elemRestriction->setupFrom(r);
+    }
+
+    ElemRestriction* ElemRestriction::from(CeedOperatorField operatorField) {
+      int ierr;
+      CeedElemRestriction ceedElemRestriction;
+
+      ierr = CeedOperatorFieldGetElemRestriction(operatorField, &ceedElemRestriction);
+      CeedOccaFromChk(ierr);
+
+      return from(ceedElemRestriction);
+    }
+
+    ElemRestriction* ElemRestriction::setupFrom(CeedElemRestriction r) {
+      int ierr;
+
+      ierr = CeedElemRestrictionGetNumElements(r, &ceedElementCount);
+      CeedOccaFromChk(ierr);
+
+      ierr = CeedElemRestrictionGetElementSize(r, &ceedElementSize);
+      CeedOccaFromChk(ierr);
+
+      ierr = CeedElemRestrictionGetNumComponents(r, &ceedComponentCount);
+      CeedOccaFromChk(ierr);
+
+      ierr = CeedElemRestrictionGetLVectorSize(r, &ceedLVectorSize);
+      CeedOccaFromChk(ierr);
+
+      // Find what type of striding the restriction uses
+      bool isStrided = false;
+      bool hasBackendStrides = false;
+
+      ierr = CeedElemRestrictionGetStridedStatus(r, &isStrided);
+      CeedOccaFromChk(ierr);
+
+      if (isStrided) {
+        ierr = CeedElemRestrictionGetBackendStridesStatus(r, &hasBackendStrides);
+        CeedOccaFromChk(ierr);
+      }
+
+      if (isStrided) {
+        if (hasBackendStrides) {
+          ceedStrideType = FULLY_STRIDED;
+        } else {
+          ceedStrideType = NOT_STRIDED;
+        }
+      } else {
+        ceedStrideType = COMPONENT_STRIDED;
+      }
+
+      if (ceedStrideType == FULLY_STRIDED) {
+        CeedInt strides[3];
+
+        ierr = CeedElemRestrictionGetStrides(r, &strides);
```

Comment:
`CeedElemRestrictionGetStrides()` should be called when `CeedElemRestrictionGetBackendStridesStatus(r, &hasBackendStrides)` is false. The logic is 'if the using the backend strides, use whatever the backend prefers (lines 273-275), otherwise use the user provided strides'.